hammr/commands/image/image.py

```diff
@@ -197,6 +198,99 @@ def help_publish(self):
 
+    def arg_deploy(self):
+        doParser = ArgumentParser(prog=self.cmd_name + " deploy", add_help=True,
+                                  description="Deploy an instance of a published image on the targeted cloud.")
+        mandatory = doParser.add_argument_group("mandatory arguments")
+        mandatory.add_argument('--id', dest='pid', required=True,
+                               help="the ID of the published image to deploy")
+        mandatory.add_argument('-n', '--name', dest='deploy_name', required=True,
+                               help="the name of the image to deploy")
+
+        optional = doParser.add_argument_group("optional arguments")
+        optional.add_argument('--vcpu', dest='vcpu', required=False,
+                              help="minimal number of cores for the image to deploy")
+        optional.add_argument('-m', '--memory', dest='memory', required=False,
+                              help="minimal RAM for the image to deploy")
+        return doParser
+
+    def do_deploy(self, args):
+        try:
+            # add arguments
+            doParser = self.arg_deploy()
+            doArgs = doParser.parse_args(shlex.split(args))
+
+            # if the help command is called, parse_args returns None object
+            if not doArgs:
+                return 2
+
+            if doArgs.pid and doArgs.deploy_name:
+
+                pimage = self.get_pimage_from_id(doArgs.pid)
+                if pimage == 2:
+                    return
+
+                image_id = generics_utils.extract_id(pimage.imageUri)
+                if image_id is None or image_id == "":
+                    printer.out("Image not found", printer.ERROR)
+                    return 2
+
+                appliance = self.api.Users(self.login).Appliances(generics_utils.extract_id(pimage.applianceUri)).Get()
+                if appliance is None or not hasattr(appliance, 'dbId'):
+                    printer.out("No template found for this image", printer.ERROR)
+                    return
+
+                if not self.is_pimage_ready_to_deploy(pimage):
+                    printer.out("Published image with name '" + pimage.name + " cannot be deployed", printer.ERROR)
+                    return 2
+
+                deployment = self.get_deployment_from_args_for_deploy(doArgs)
+                deployed_instance = self.api.Users(self.login).Appliances(appliance.dbId).Images(image_id).Pimages(
+                    pimage.dbId).Deploys.Deploy(body=deployment, element_name="ns1:deployment")
+                deployed_instance_id = deployed_instance.applicationId
+
+                print("Deployment in progress")
+
+                status = self.api.Users(self.login).Deployments(deployed_instance_id).Status.Getdeploystatus()
+                bar = ProgressBar(widgets=[BouncingBar()], maxval=UnknownLength)
+                bar.start()
+                i = 1
+                while not (status.message == "running" or status.message == "on-fire"):
```

Comment:
While not message == running ou success ?
I don't understand why you stop put the progress bar to finish when the status is running....